extensions/github-authentication/src/github.ts

```diff
@@ -31,6 +32,20 @@ export enum AuthProviderType {
 	githubEnterprise = 'github-enterprise'
 }
 
+interface GitHubAuthenticationProviderOptions extends vscode.AuthenticationProviderSessionOptions {
+	/**
+	 * This is specific to GitHub and is used to determine which social sign-in provider to use.
+	 * If not provided, the default (GitHub) is used which shows all options.
+	 *
+	 * Example: If you specify Google, then the sign-in flow will skip the initial page that asks you
+	 * to choose how you want to sign in and will directly take you to the Google sign-in page.
+	 *
+	 * This allows us to show "Continue with Google" buttons in the product, rather than always
+	 * leaving it up to the user to choose the social sign-in provider on the sign-in page.
+	 */
+	readonly provider?: GitHubSocialSignInProvider;
```

Comment:
@bpasero since our auth API aligns with the OAuth spec and this whole Google sign in flow does not align with that, we have to do something custom like this.

I expect you to pass in `provider: 'google'` in the call to `createSession`